tests/integration/targets/vmware_ops_esxi_maintenance_mode_test/tasks/main.yml

```diff
@@ -0,0 +1,106 @@
+---
+- name: Test for esxi maintenance on simulator
+  tags: integration-ci
+  block:
+    - name: Enable Maintenance Mode
+      ansible.builtin.include_role:
+        name: cloud.vmware_ops.esxi_maintenance_mode
+      vars:
+        esxi_maintenance_mode_enable: True
```

Comment:
I would add a task to assert that the esxi host is in maintenance mode, as well as after disabling maintenance mode to check it is disabled